### PR TITLE
Fix cppcheck static analysis errors

### LINKED ISSUE #15018: Errors in the static analysis of cppcheck
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.0-rev | Wazuh shared libraries | Agent | Sources | All |

## Description
Due to the increase of the OSVersion in the docker image that executes the RTR -> issue: https://github.com/wazuh/wazuh-jenkins/issues/3921, the version of cppcheck was also updated, which causes new things to be detected and also allows for to eliminate of the suppressions due to existing bugs in past versions.

Although this issue does not touch anything functional, it is crucial for 4.4, since it favors getting the full green of the checks.

Errors:
DataProvider
```
/home/dwordcito/Development/wazuh/src/data_provider/src/sysInfo.cpp:14:1: error: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter' [ctuOneDefinitionRuleViolation]
struct CJsonDeleter
^
/home/dwordcito/Development/wazuh/src/data_provider/tests/sysInfo/sysInfo_test.cpp:96:1: note: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter'
struct CJsonDeleter final
^
/home/dwordcito/Development/wazuh/src/data_provider/src/sysInfo.cpp:14:1: note: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter'
struct CJsonDeleter
```
DBSync:
```
/home/dwordcito/Development/wazuh/src/shared_modules/dbsync/integrationTests/fim/fimIntegrationTest.cpp:30:1: error: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter' [ctuOneDefinitionRuleViolation]
struct CJsonDeleter final
^
/home/dwordcito/Development/wazuh/src/shared_modules/dbsync/src/dbsync.cpp:25:1: note: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter'
struct CJsonDeleter
^
/home/dwordcito/Development/wazuh/src/shared_modules/dbsync/integrationTests/fim/fimIntegrationTest.cpp:30:1: note: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter'
struct CJsonDeleter final
^
```

RSYNC:
```
/home/dwordcito/Development/wazuh/src/shared_modules/rsync/src/rsyncImplementation.h:25:1: error: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter' [ctuOneDefinitionRuleViolation]
struct CJsonDeleter
^
/home/dwordcito/Development/wazuh/src/shared_modules/rsync/testtool/agentEmulator.cpp:44:1: note: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter'
struct CJsonDeleter final
^
/home/dwordcito/Development/wazuh/src/shared_modules/rsync/src/rsyncImplementation.h:25:1: note: The one definition rule is violated, different classes/structs have the same name 'CJsonDeleter'
struct CJsonDeleter
^
/home/dwordcito/Development/wazuh/src/shared_modules/rsync/src/rsyncImplementation.cpp:273:25: error: Using argument ctx that points at uninitialized variable checksumCtx [ctuuninitvar]
    const auto middle { ctx.size / 2 };
                        ^
/home/dwordcito/Development/wazuh/src/shared_modules/rsync/src/rsyncImplementation.cpp:90:29: note: Calling function fillChecksum, 5th argument is uninitialized
                fillChecksum(spDBSyncWrapper, jsStartParams, begin, end, checksumCtx);
                            ^
/home/dwordcito/Development/wazuh/src/shared_modules/rsync/src/rsyncImplementation.cpp:273:25: note: Using argument ctx
    const auto middle { ctx.size / 2 };
```

## DoD
- [x] Remove suppressions.
- [x] Fix all cppcheck warnings.

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|Closes #15018 |

## Description
This PR aims to resolve different warnings that appeared after updating the cppcheck version in the RTR docker containers.

In the change, the structures are templatized and given a common use, avoiding their redefinition, and also defining a single approach for the use of this c++ characteristic in the use of Customs Deleters.

another change is the initialization of a structure that was not initialized.

and another more trivial change that must be done is the elimination of the suppression files, as well as the command line of the tool that indicated the file path of that file.

### CODE REVIEW

File: src/shared_modules/rsync/testtool/agentEmulator.cpp [hunk 1]
```diff
@@ -33,27 +33,11 @@ static DBSYNC_HANDLE createDbsyncHandle(const std::string& dbName)
...
-        cJSON_free(json);
-    }
-};
-
 static void callback(const ReturnTypeCallback /*type*/,
                      const cJSON* /*json*/,
                      void* /*ctx*/)
 {
-    // const std::unique_ptr<char, CJsonDeleter> spJsonBytes{ cJSON_PrintUnformatted(json) };
+    // const std::unique_ptr<char, CJsonSmartFree> spJsonBytes{ cJSON_PrintUnformatted(json) };
```
Comment (MiguelazoDS) [inline src/shared_modules/rsync/testtool/agentEmulator.cpp:40]: What is this callback function used for?

File: src/shared_modules/rsync/tests/interface/rsync_test.cpp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include "rsync.hpp"
 #include "dbsync.h"
 #include "dbsync.hpp"
+#include "cjsonSmartDeleter.hpp"
```
Comment (MiguelazoDS) [inline src/shared_modules/rsync/tests/interface/rsync_test.cpp:20]: There's an unused struct CJsonDeleter in this file https://github.com/wazuh/wazuh/blob/15018-cppcheck-static-analysis-errors/src/shared_modules/rsync/tests/interface/rsync_test.cpp#L66

File: src/shared_modules/rsync/testtool/agentEmulator.cpp (hunk 1 above)
Comment (Dwordcito) [reply src/shared_modules/rsync/testtool/agentEmulator.cpp:40]: this callback is for debugging only.

File: src/shared_modules/rsync/tests/interface/rsync_test.cpp (hunk 2 above)
Comment (Dwordcito) [reply src/shared_modules/rsync/tests/interface/rsync_test.cpp:20]: good catch!

**Review by MiguelazoDS — APPROVED** [review]

**Review by jmv74211 — APPROVED** [review]
## QA review

- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3372
- **Comments**:  https://github.com/wazuh/wazuh-qa/issues/3372#issuecomment-1274895285

**Pull request merged**